components/calendar/src/islamic.rs

```diff
@@ -559,7 +570,11 @@ impl IslamicUmmAlQura {
         let month = ((div_rem_euclid64(elapsed_months, 12).1) + 1) as u8;
         let day = ((date - crescent) + 1) as u8;
 
-        Date::try_new_ummalqura_date(year, month, day).unwrap()
+        debug_assert!(Date::try_new_ummalqura_date(year, month, day).is_ok());
```

Comment:
Nit (here and elsewhere): Remove the `#[allow(clippy::unwrap_used)]`

If we don't think we should be unwrapping here, we should probably do something similar in other places where we have similar code in other calendars.

Reply:
I think I'll merge this because I plan to do a pass through these allows in the new calendars tomorrow, and I'll make sure to catch it then.

Reply:
(just pointing out that there are two parts of my comment, one about other calendars and one about clippy)

Reply:
yep, I'll be cleaning up both!